### DISCUSSION TITLE
use override file name in config

### DISCUSSION THREAD

**Discussion by qsh-zh** [facebookresearch/hydra#2299] [Q&A] [answered]
Here is a toy example with the following config tree
```
├── config.yaml
├── net
│   ├── cnn.yaml
│   └── mlp.yaml
└── optimizer
    ├── adam.yaml
    └── sgd.yaml
```

And `config.yaml` reads

```yaml
defaults:
  - net: cnn
  - optimizer: sgd
 
name: cnn_sgd
```

and example of `cnn.yaml`
```yaml
_target_: SomeFancyCNN
num_layer: 3
```
# Question

In the above example, we have totally 4 combinations~([cnn, mlp] x [sgd, adam]). Do we have a smart and automatic way to set `name`, for example, 
```
python run.py net=mlp optimizer=adam # config.name -> mlp_adam
python run.py net=mlp optimizer=sgd # config.name -> mlp_sgd
python run.py net=cnn optimizer=sgd # config.name -> cnn_sgd
```

**Comment by Jasha10** [comment, marked as answer]
A few possibilities come to mind:
- You could use [`${hydra:runtime.choices}`](https://hydra.cc/docs/next/configure_hydra/intro/#hydraruntime) to get a list of config group choices. This list could be further processed by a [custom resolver](https://omegaconf.readthedocs.io/en/latest/custom_resolvers.html) to turn the list into a string.
- [`${hydra:job.override_dirname}`](https://hydra.cc/docs/next/configure_hydra/intro/#hydrajob) is a convenience providing a ready-to-use string representing the overrides used for the current job.
